### ISSUE TITLE
TS file that doesn't even generate an output file

### ISSUE THREAD

**Issue by cfsmp3** [CCExtractor/ccextractor#385]
The file friends_tbs.ts, which was just shared in slack and will be uploaded to the global repository, doesn't generate any output. I mean, it doesn't even generate a .srt file using this:

/home/captions/ccextractorgit/ccextractor/linux/ccextractor -srt friends_tbs.ts

The output is 
Opening file: friends_tbs.ts
File seems to be a transport stream, enabling TS mode
Analyzing data in general mode
Done, processing time = 0 seconds
This is beta software. Report issues to carlos at ccextractor org...

It processes it instantly and produces no output file. 
VLC plays it correctly.

I haven't analyzed it myself yet because there's a bunch of other things I'm doing right now but this probably evidences a really weird bug we need to look into asap.

**Comment by cfsmp3** [comment]
Assigned to all veterans

**Comment by canihavesomecoffee** [comment]
Ran it with -parsedebug

Prints "Packet (pid 5355, 5357, 5358, 5389) skipped - no stream with captions identified yet" until it runs out of file to process.

Interestingly VLC seems to be one of the only ones that can display the captions...

* MediaInfo can't detect any subs in here
* MPC-HC detects no captions

**Comment by cfsmp3** [comment]
Pasting my own rant from slack:

So what I'm seeing is that we aren't parsing the PMT and we are leaving the function here:

if (table_id == 0xC0)
    {
        /*
         * Acc to System Information for Satellite Distribution
         * of Digital Television for Cable and MMDS (ANSI/SCTE 57 2003 )
         * PROGRAM INFORMATION Table found in PMT
         */
        dbg_print(CCX_DMT_PARSE, "PMT: PROGRAM INFORMATION Table need implementation");
        return 0;
    }

[a bit later]
So good news is that the regular PMT comes after that section
so I got it to work by cheating that removing that section from the buffer

But bad news is that we should be processing the PMT without making assumptions about sections in the PMT... for example now I'm just deleting that from the buffer (not skipping it, because the parsing code assumes that the usual table starts at the beginning of the buffer)

but we should have a loop that checks the table_id, if it's 0xC0 skips it (or better, parse it, even though we don't need it for now just to get the captions), if it's 2 parses it as usual..... and then processes the next section without assuming order of sections, as apparently they may come in any order

**Comment by cfsmp3** [comment]
Direct link:
https://drive.google.com/open?id=0B_61ywKPmI0TRm9iM2pBbDZPOWc

**Comment by grave-w-grave** [comment]
Confirming that this is fixed as of 0.84.

```
CCExtractor 0.84, Carlos Fernandez Sanz, Volker Quetschke.
Teletext portions taken from Petr Kutalek's telxcc
--------------------------------------------------------------------------
Input: friends_tbs.ts
[Extract: 1] [Stream mode: Autodetect]
[Program : Auto ] [Hauppage mode: No] [Use MythTV code: Auto]
[Timing mode: Auto] [Debug: No] [Buffer input: Yes]
[Use pic_order_cnt_lsb for H.264: No] [Print CC decoder traces: No]
[Target format: .srt] [Encoding: UTF-8] [Delay: 0] [Trim lines: No]
[Add font color data: Yes] [Add font typesetting: Yes]
[Convert case: No] [Video-edit join: No]
[Extraction start time: not set (from start)]
[Extraction end time: not set (to end)]
[Live stream: No] [Clock frequency: 90000]
[Teletext page: Autodetect]
[Start credits text: None]

-----------------------------------------------------------------
Opening file: friends_tbs.ts
File seems to be a transport stream, enabling TS mode
Analyzing data in general mode
Creating friends_tbs.srt
  1%  |  00:00

New video information found
[704 * 480] [AR: 02 - 4:3] [FR: 04 - 29.97] [progressive: no]

  2%  |  00:01XDS Notice: Network is now TBS
XDS: ContentAdvisory: US TV Parental Guidelines. Age Rating: TV-PG (Parental Guidance Suggested)
XDS:
XDS Program Type: [21-Entertainment]
XDS Notice: Program is now Friends (SAP), One Where Rachel
XDS: CGMS: One generation of copies can be made
XDS: APS: No APS
XDS: Redistribution Control Descriptor: 0
100%  |  01:37
Number of NAL_type_7: 0
Number of VCL_HRD: 0
Number of NAL HRD: 0
Number of jump-in-frames: 0
Number of num_unexpected_sei_length: 0

Total frames time:        00:01:36:729  (2899 frames at 29.97fps)

Min PTS:                                22:10:12:815
Max PTS:                                22:11:50:880
Length:                          00:01:38:065

Initial GOP time:          00:00:00:000
Final GOP time:          00:00:00:000+33F
Diff. GOP length:          00:00:00:000+33F     (00:00:01:101)


Total user data fields: 5798
SCTE-20 type user data fields: 2899
Done, processing time = 0 seconds
This is beta software. Report issues to carlos at ccextractor org...
```
[friends_tbs.zip](https://github.com/CCExtractor/ccextractor/files/691199/friends_tbs.zip)
